File: Mario1.java
import java.util.Scanner;

public class Mario1{
    public static void main(String[]args){
        Scanner scanner = new Scanner(System.in);
        System.out.print("Quantos blocos de pirÂmide vai querer fazer?: ");
        int vezes = scanner.nextInt();
        int i = 1;
      do { 
          String estrela = "#".repeat(i);
          System.out.println(estrela);
          i++;
      } while (i <= vezes);
      scanner.close();
    }
}

File: mario1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(){
    int vezes;
    printf("Quanto de pirâmide vai querer fazer? : ");
    scanf("%d", &vezes);

    int i = 1;
    do{
        char estrela[vezes + 1];
        memset(estrela, '#', i);
        estrela[i] = '\0';

        printf("%s\n", estrela);
        i++;
    }while(i <= vezes);
    return 0;
}